### DISCUSSION TITLE
Web Worker not working

### DISCUSSION THREAD

**Discussion by stoplion** [webpack/webpack#13224] [General]
Using Webpack 5.33.2.
Following instructions here..
https://webpack.js.org/guides/web-workers/#root

```
const workerModuleUrl = '../../workers/worker.ts';
worker = new Worker(new URL(workerModuleUrl, import.meta.url));
```

```
    worker.postMessage({
      payload: value,
    });

    worker.onmessage = () => {
      alert('worker working..');
    };
```

`Security Error: Content at http://localhost:5000/@tester/ea-ut-eum-at may not load data from file:///Users/mycomp/Code/MyApp/app/client/src/workers/worker.ts.`

**Comment by stoplion** [comment]
My app really depends on web workers. I'm using Rails, so the default bundler is Webpacker. The docs give the above instructions, and not much else is explained. Am I missing something here?

**Reply by alexander-akait** [reply]
What is `target`? Maybe you can create simple example? Do you use SSR?

**Comment by sokra** [comment]
The whole expression of `new Worker(new URL(..., import.meta.url))` must be statically analyse-able. You must inline the string into the expression:

``` js
worker = new Worker(new URL('../../workers/worker.ts', import.meta.url));
```
